### PR TITLE
name validation

### PR DESCRIPTION
**Pull request by harshavardhanunta**
Bug description : Name column allows the user adding special characters and numbers also 
Root cause : They are not validated the name column and pushing to data base  
Fix : Added  the logic for name validation

### CODE REVIEW

File: employee_project/employee.py [hunk 1]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
         print("Invalid input, enter a valid input ex: 20000")
         sal = get_emp_salary_input(msg)
     return sal
+
+
+def employee_name(employee_name):
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: Change the method employee_name to get_employee_name(employee_name)

File: employee_project/employee.py [hunk 2]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+    """ @param: string
+        @desc: method returns the valide name
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: change the method description:  This Method receives the input from the user and validating the name .

File: employee_project/employee.py [hunk 3]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+    while True:
+        Name_min = 3
+        Name_max = 15
+        employee_name = input("Enter the  employee_name:")
+        length_name = len(employee_name)
+        if employee_name.isalpha() == True or ' ' in employee_name:
+            if (Name_min < length_name and Name_max > length_name):
+                emp_data = employee_name
+                return emp_data
+                break
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: break statement is not required.

File: employee_project/employee.py [hunk 4]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+        @returns: string
+    """
+    while True:
+        Name_min = 3
+        Name_max = 15
+        employee_name = input("Enter the  employee_name:")
+        length_name = len(employee_name)
+        if employee_name.isalpha() == True or ' ' in employee_name:
+            if (Name_min < length_name and Name_max > length_name):
+                emp_data = employee_name
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: No need emp_data variable here , Just return employee_name

File: employee_project/employee.py [hunk 5]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
     return sal
+
+
+def employee_name(employee_name):
+    """ @param: string
+        @desc: method returns the valide name
+        @returns: string
+    """
+    while True:
+        Name_min = 3
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: Follow the naming conventions , Name_min 
Change Name_min to min_lenth_name

File: employee_project/employee.py [hunk 6]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+        Name_max = 15
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: Follow the naming conventions , Name_max 
Change Name_max  to max_lenth_name

File: employee_project/employee.py [hunk 7]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+        employee_name = input("Enter the  employee_name:")
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: Don't give the description employee_name like this 

"Enter the  Employee Name:"

File: employee_project/employee.py [hunk 8]
```diff
@@ -86,3 +86,24 @@ def get_emp_salary_input(msg = "Enter employee salary:"):
...
+        length_name = len(employee_name)
+        if employee_name.isalpha() == True or ' ' in employee_name:
+            if (Name_min < length_name and Name_max > length_name):
+                emp_data = employee_name
+                return emp_data
+                break
+            else:
+                print("Enter the minmum 3 chacters and maximun 15 characters :")
+        else:
+            print("provide the valide name")
```
Comment (SuryaBhai-Chinna) [inline employee_project/employee.py]: "Invalid name entered , Please provide the valid name"

File: employee_project/main.py [hunk 9]
```diff
@@ -11,7 +11,7 @@ def main():
         if choice == 1:
             os.system("cls")
             print("***************** REGISTER EMPLOYEE DETAILS ******************************\n\n")
-            name = input("Enter Employee Name: ")
+            name = employee_name(input("Enter Employee Name: "))
```
Comment (SuryaBhai-Chinna) [inline employee_project/main.py]: Why your adding here also input and employee.py file also adding the same line to get the input from the user.
employee_name(input("Enter Employee Name: "))

File: employee_project/employee.py (hunk 1 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: Changeded the method employee_name to get_employee_name(employee_name)

File: employee_project/employee.py (hunk 2 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: changed the method description

File: employee_project/employee.py (hunk 3 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: removed the break

File: employee_project/employee.py (hunk 4 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: changed

File: employee_project/employee.py (hunk 5 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: changed

File: employee_project/employee.py (hunk 6 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: changed

File: employee_project/employee.py (hunk 7 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: removed from the main.py

File: employee_project/employee.py (hunk 8 above)
Comment (harshavardhanunta) [reply employee_project/employee.py]: changed

File: employee_project/main.py (hunk 9 above)
Comment (harshavardhanunta) [reply employee_project/main.py]: removed

**Review by SuryaBhai-Chinna — APPROVED** [review]

**Pull request merged**